specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2024-05-15-preview/networkSecurityPerimeter.json

```diff
@@ -0,0 +1,455 @@
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "A collection of Network Security Perimeter configurations",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkSecurityPerimeterConfiguration"
+          }
+        },
+        "nextLink": {
+          "description": "The link used to get the next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "NetworkSecurityPerimeterConfigurationProperties": {
+      "description": "Properties of the Network Security Perimeter Configuration",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning state of Network Security Perimeter configuration propagation",
+          "enum": [
+            "Accepted",
+            "Succeeded",
+            "Failed",
+            "Deleting",
+            "Canceled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "NetworkSecurityPerimeterConfigurationProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "provisioningIssues": {
+          "description": "List of Provisioning Issues if any",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/ProvisioningIssue"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "networkSecurityPerimeter": {
+          "$ref": "#/definitions/NetworkSecurityPerimeter",
+          "readOnly": true
+        },
+        "resourceAssociation": {
+          "description": "Information about resource association",
+          "readOnly": true,
+          "type": "object",
+          "properties": {
+            "name": {
+              "description": "Name of the resource association",
+              "type": "string"
+            },
+            "accessMode": {
+              "description": "Access Mode of the resource association",
+              "enum": [
+                "Enforced",
+                "Learning",
+                "Audit"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "ResourceAssociationAccessMode",
+                "modelAsString": true
+              }
+            }
+          }
+        },
+        "profile": {
+          "description": "Network Security Perimeter profile",
+          "readOnly": true,
+          "type": "object",
+          "properties": {
+            "name": {
+              "description": "Name of the resource",
+              "type": "string"
+            },
+            "accessRulesVersion": {
+              "description": "Current access rules version",
+              "type": "number"
```

Comment:
> "type": "number" [](http://example.com/codeflow?start=13&length=17)

Bot Service swagger thinks this is an int64. Whereas you've gone for number, which allows floating point. Who is correct, you or them?

```
        "accessRulesVersion": {
          "description": "Current access rules version",
          "type": "integer",
          "format": "int64"
        },
```

Reply:
It should be a 32-bit integer as per NRP design.

Reply:
Thanks for checking. :D